hphp/runtime/vm/jit/vasm-ppc64.cpp

```diff
@@ -1022,13 +1013,27 @@ void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
   else v << vasm_dst_imm{inst.s0, tmp2, inst.sf};                       \
 }
 
-X(cmpbim,  cmpl,  cmpli,  loadb)
+X(cmpbim,  cmpq,  cmpqi,  loadzbq)
 X(cmpqim,  cmpq,  cmpqi,  load)
-X(testbim, testq, testqi, loadb)
-X(testwim, testq, testqi, loadw)
-X(testlim, testq, testqi, loadl)
+X(testbim, testq, testqi, loadzbq)
 X(testqim, testq, testqi, load)
+#undef X
+
+#define X(vasm_src, vasm_dst_reg, vasm_dst_imm, vasm_load, vasm_exts)   \
+void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
+  Vptr p = inst.s1;                                                     \
+  Vreg tmp; if (!e.allow_vreg()) tmp = Vreg(PhysReg(rAsm));             \
```

Comment:
Rewrite this as a ternary.

Reply:
I have removed the tmp set, as it is already made in ```patchImm``` function.